backend/functions/tasks.py

```diff
@@ -0,0 +1,206 @@
+        columns=[
+            "sentence_text_id",
+            "corrected_text",
+            "input_language",
+            "context",
+            "quality_status",
+            "metadata",
+        ],
+    )
+
+    # Keep only the clean sentences
+    input_sentences_df = input_sentences_df[
+        (input_sentences_df["quality_status"] == "Clean")
+    ].reset_index(drop=True)
+    # Check if the dataframe is empty
+    if input_sentences_df.shape[0] == 0:
+
+        # Update the task status
+        self.update_state(
+            state="FAILURE",
+            meta={
+                "No sentences to upload.",
+            },
+        )
+
+        raise Exception("No clean sentences found. Perform project export first.")
+
+    # Make a sentence list for valid sentences to be translated
+    all_sentences_to_be_translated = input_sentences_df["corrected_text"].tolist()
+
+    # Get the output dataset instance
+    output_dataset_instance = dataset_models.DatasetInstance.objects.get(
+        instance_id=output_dataset_instance_id
+    )
+    # Create a TranslationPair object list
+    translation_pair_objects = []
+
+    # Iterate through the languages
+    for output_language in languages:
+
+        # Loop through all the sentences to be translated in batch format
+        for i in range(0, len(all_sentences_to_be_translated), batch_size):
+
+            batch_of_input_sentences = all_sentences_to_be_translated[
+                i : i + batch_size
+            ]
+
+            # Check the API type
+            if api_type == "indic-trans":
+
+                # Get the translation using the Indictrans NMT API
+                translations_output = get_batch_translations_using_indictrans_nmt_api(
+                    batch_of_input_sentences,
+                    input_sentences_df["input_language"].iloc[0],
+                    output_language,
+                    checks_for_particular_languages=checks_for_particular_languages,
+                )
+
+                # Check if the translations output is a string or a list
+                if isinstance(translations_output, str):
+
+                    # Update the task status and raise an exception
+                    self.update_state(
+                        state="FAILURE",
+                        meta={
+                            "Error: {}".format(translations_output),
+                        },
+                    )
+
+                    raise Exception(translations_output)
+                else:
+                    translated_sentences = translations_output
+
+            elif api_type == "google":
+                # Get the translation using the Indictrans NMT API
+                translations_output = get_batch_translations_using_google_translate(
+                    sentence_list=batch_of_input_sentences,
+                    target_language=output_language,
+                    checks_for_particular_languages=checks_for_particular_languages,
+                )

```

Comment:
Sometimes, Google API fails to auto-detect language. In such cases, it just returns the input sentences as it is. We observed it in a few Shoonya tasks.

Can you please also explicit pass the input_langauge

Reply:
I have made the change. The `source_language` is being passed explicitly now.